### PR TITLE
fix(add-aws-payloads): add SNS and SQS payloads

### PR DESCRIPTION
**Pull request by haddasbronfman**

### CODE REVIEW

**Review by nemoshlag — APPROVED** [review]

File: src/instrumentations/extentions/aws/DynamoDBEventCreator.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  AwsSdkResponseHookInformation,
+} from '@opentelemetry/instrumentation-aws-sdk';
+import { AWSEventCreator } from './event-creator-interface';
+
+export class DynamoDBEventCreator implements AWSEventCreator {
+  requestHandler(span: Span, requestInfo: AwsSdkRequestHookInformation): void {
+    switch (requestInfo.request.commandName) {
+      case 'PutItem':
+        span.setAttribute(
+          'Table Name',
```
Comment (osherv) [inline src/instrumentations/extentions/aws/DynamoDBEventCreator.ts:28]: attribute labels should be followed otel semantic conventions:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md

File: src/instrumentations/extentions/aws/DynamoDBEventCreator.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          requestInfo.request.commandInput.TableName
+        );
+        span.setAttribute(
+          'Item',
```
Comment (osherv) [inline src/instrumentations/extentions/aws/DynamoDBEventCreator.ts:32]: same here:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md

File: src/instrumentations/extentions/aws/DynamoDBEventCreator.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        break;
+    }
+  }
+  responseHandler(
+    span: Span,
+    responseInfo: AwsSdkResponseHookInformation
+  ): void {
+    switch (responseInfo.response.request.commandName) {
+      case 'PutItem':
+        span.setAttribute('data', JSON.stringify(responseInfo.response.data));
```
Comment (osherv) [inline src/instrumentations/extentions/aws/DynamoDBEventCreator.ts:44]: same here
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md

File: src/instrumentations/extentions/aws/SQSEventCreator.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  AwsSdkRequestHookInformation,
+  AwsSdkResponseHookInformation,
+} from '@opentelemetry/instrumentation-aws-sdk';
+import { AWSEventCreator } from './event-creator-interface';
+
+export class SQSEventCreator implements AWSEventCreator {
+  requestHandler(span: Span, requestInfo: AwsSdkRequestHookInformation): void {
+    const queueUrl = requestInfo.request.commandInput.QueueUrl;
+    const splited = queueUrl.split('/');
+    span.setAttribute('aws.sqs.queue_name', splited.pop());
```
Comment (osherv) [inline src/instrumentations/extentions/aws/SQSEventCreator.ts:27]: are this always returns array of the experted field values?

File: src/instrumentations/extentions/aws/SQSEventCreator.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        );
+        break;
+      case 'ReceiveMessage': {
+        if (responseInfo.response.data.Messages.length == 1) {
+          span.setAttribute(
+            'aws.sqs.record.message_body',
+            responseInfo.response.data.Messages[0].Body
+          );
+          span.setAttribute(
+            'aws.sqs.record.message_attributes',
```
Comment (osherv) [inline src/instrumentations/extentions/aws/SQSEventCreator.ts:96]: should follow https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md.

'message.name.label1', 'message.name.label2'

File: src/instrumentations/extentions/aws/SQSEventCreator.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+          );
+          span.setAttribute(
+            'aws.sqs.record.attributes.aws_trace_header',
+            responseInfo.response.data.Messages[0].Attributes.AWSTraceHeader
+          );
+        }
+        if (responseInfo.response.data.Messages.length > 1) {
+          span.setAttribute(
+            'aws.sqs.record',
+            JSON.stringify(responseInfo.response.data.Messages)
```
Comment (osherv) [inline src/instrumentations/extentions/aws/SQSEventCreator.ts:121]: same https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md.

The attributes labels should be flat

File: src/instrumentations/extentions/aws/SQSEventCreator.ts [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        span.setAttribute(
+          'aws.sqs.record',
+          JSON.stringify(responseInfo.response.data)
+        );
+        break;
+      case 'ReceiveMessage': {
+        if (responseInfo.response.data.Messages.length == 1) {
+          span.setAttribute(
+            'aws.sqs.record.message_body',
+            responseInfo.response.data.Messages[0].Body
```
Comment (osherv) [inline src/instrumentations/extentions/aws/SQSEventCreator.ts:93]: same https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md.

The attributes labels should be flat

File: src/instrumentations/extentions/aws/SQSEventCreator.ts [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+    const queueUrl = requestInfo.request.commandInput.QueueUrl;
+    const splited = queueUrl.split('/');
+    span.setAttribute('aws.sqs.queue_name', splited.pop());
+    span.setAttribute('aws.account_id', splited.pop());
+    // span.setAttribute('aws.region', `${requestInfo.request.region}`);
+
+    switch (requestInfo.request.commandName) {
+      case 'SendMessage':
+        span.setAttribute(
+          'aws.sqs.record.message_body',
```
Comment (osherv) [inline src/instrumentations/extentions/aws/SQSEventCreator.ts:34]: same https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md.

The attributes labels should be flat

**Review by osherv — CHANGES_REQUESTED** [review]

**Comment by osherv** [comment]
Hey! Looks rly great! I've added few comments.
And two more things:
1. We should choose one convention for file names (one of (this-conection, this_convention, ThisConvention, thisConvention)
2. we should understand with Yossef, Nadav our new metadata_only using, maxPayload

File: src/instrumentations/extentions/aws/SQSEventCreator.ts [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+        );
+        break;
+      case 'SendMessageBatch':
+        span.setAttribute(
+          'aws.sqs.entries',
+          JSON.stringify(requestInfo.request.commandInput.Entries)
+        );
+        break;
+      case 'ReceiveMessage':
+        // span.setAttribute(
```
Comment (osherv) [inline src/instrumentations/extentions/aws/SQSEventCreator.ts:53]: remove or uncomment

File: src/instrumentations/extentions/aws/SQSEventCreator.ts [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+    // span.setAttribute('aws.region', `${requestInfo.request.region}`);
+
+    switch (requestInfo.request.commandName) {
+      case 'SendMessage':
+        span.setAttribute(
+          'aws.sqs.record.message_body',
+          requestInfo.request.commandInput.MessageBody
+        );
+        // span.setAttribute(
+        //   'aws.sqs.record.message_attributes',
```
Comment (osherv) [inline src/instrumentations/extentions/aws/SQSEventCreator.ts:38]: remove or uncomment

File: src/instrumentations/extentions/aws/SQSEventCreator.ts [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+} from '@opentelemetry/instrumentation-aws-sdk';
+import { AWSEventCreator } from './event-creator-interface';
+
+export class SQSEventCreator implements AWSEventCreator {
+  requestHandler(span: Span, requestInfo: AwsSdkRequestHookInformation): void {
+    const queueUrl = requestInfo.request.commandInput.QueueUrl;
+    const splited = queueUrl.split('/');
+    span.setAttribute('aws.sqs.queue_name', splited.pop());
+    span.setAttribute('aws.account_id', splited.pop());
+    // span.setAttribute('aws.region', `${requestInfo.request.region}`);
```
Comment (osherv) [inline src/instrumentations/extentions/aws/SQSEventCreator.ts:29]: remove or uncomment

**Review by osherv — CHANGES_REQUESTED** [review]

File: src/instrumentations/extentions/aws/DynamoDBEventCreator.ts (hunk 1 above)
Comment (haddasbronfman) [reply src/instrumentations/extentions/aws/DynamoDBEventCreator.ts:28]: added "TODO" to handle this later. We should go here by the specification

File: src/instrumentations/extentions/aws/DynamoDBEventCreator.ts (hunk 3 above)
Comment (haddasbronfman) [reply src/instrumentations/extentions/aws/DynamoDBEventCreator.ts:44]: added "TODO" to handle this later. We should go here by the specification

File: src/instrumentations/extentions/aws/SQSEventCreator.ts (hunk 5 above)
Comment (haddasbronfman) [reply src/instrumentations/extentions/aws/SQSEventCreator.ts:96]: We should decide about the spec and act accordingly.

File: src/instrumentations/extentions/aws/SQSEventCreator.ts (hunk 4 above)
Comment (haddasbronfman) [reply src/instrumentations/extentions/aws/SQSEventCreator.ts:27]: I'll fix this

File: src/instrumentations/extentions/aws/SQSEventCreator.ts (hunk 6 above)
Comment (haddasbronfman) [reply src/instrumentations/extentions/aws/SQSEventCreator.ts:121]: This is the case of receiving multiple messages with the same attributes. We should think how to declare it in the spec

**Pull request merged**